src/Components/Panels/ExploreLogsVizPanelMenu.tsx

```diff
@@ -0,0 +1,194 @@
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from '../../services/analytics';
+import { logger } from '../../services/logger';
+import { AddToExplorationButton } from '../ServiceScene/Breakdowns/AddToExplorationButton';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { ExtensionPoints } from '../../services/extensions/links';
+
+const ADD_TO_INVESTIGATION_MENU_TEXT = 'Add to investigation';
+const ADD_TO_INVESTIGATION_MENU_DIVIDER_TEXT = 'Add to investigation div';
+
+interface ExploreLogsVizPanelMenuState extends SceneObjectState {
+  body?: VizPanelMenu;
+  frame?: DataFrame;
+  labelName?: string;
+  fieldName?: string;
+  addToExplorations?: AddToExplorationButton;
+}
+
+/**
+ * @todo the VizPanelMenu interface is overly restrictive, doesn't allow any member functions on this class, so everything is currently inlined
+ */
+export class ExploreLogsVizPanelMenu
+  extends SceneObjectBase<ExploreLogsVizPanelMenuState>
+  implements VizPanelMenu, SceneObject
+{
+  constructor(state: Partial<ExploreLogsVizPanelMenuState>) {
+    super(state);
+    this.addActivationHandler(() => {
+      this.setState({
+        addToExplorations: new AddToExplorationButton({
+          labelName: this.state.labelName,
+          fieldName: this.state.fieldName,
+          frame: this.state.frame,
+        }),
+      });
+
+      // @todo rewrite the AddToExplorationButton
+      // Manually activate scene
+      this.state.addToExplorations?.activate();
+
+      const items: PanelMenuItem[] = [
+        {
+          text: 'Explore',
+          iconClassName: 'compass',
+          shortcut: '',
+          onClick: () => onExploreClick(this),
+        },
+      ];
+
+      this.setState({
+        body: new VizPanelMenu({
+          items,
+        }),
+      });
+
+      this.state.addToExplorations?.subscribeToState(() => {
+        subscribeToAddToExploration(this);
+      });
+    });
+  }
+
+  addItem(item: PanelMenuItem): void {
+    if (this.state.body) {
+      this.state.body.addItem(item);
+    }
+  }
+  setItems(items: PanelMenuItem[]): void {
+    if (this.state.body) {
+      this.state.body.setItems(items);
+    }
+  }
+
+  public static Component = ({ model }: SceneComponentProps<ExploreLogsVizPanelMenu>) => {
+    const { body } = model.useState();
+
+    if (body) {
+      return body && <body.Component model={body} />;
+    }
+
+    return <></>;
+  };
+}
+
+const onExploreClick = (sceneRef: SceneObject) => {
+  const indexScene = sceneGraph.getAncestor(sceneRef, IndexScene);
+  const $data = sceneGraph.getData(sceneRef);
+  let queryRunner = getQueryRunnerFromChildren($data)[0];
+
+  // If we don't have a query runner, then our panel is within a SceneCSSGridItem, we need to get the query runner from there
+  if (!queryRunner) {
+    const sceneGridItem = sceneGraph.getAncestor(sceneRef, SceneCSSGridItem);
+    const queryProvider = sceneGraph.getData(sceneGridItem);
```

Comment:
AFAIK you can do `sceneGraph.getData(sceneRef)` and it will walk up the tree to find the next SceneDataProvider. With other words, no need to do `getAncestor`